### PR TITLE
Partials

### PR DESCRIPTION
**Pull request by jphastings**
A demonstration of using partials to only send some information back from the server (keep controller logic server-side)
- Moves the component information into a props key called `react`.
- Stores current state (props) inside the `ComponentRenderer`.
- Added a recursive merge routine for partials (hat tip to [Markus](http://stackoverflow.com/a/383245/201130))

### CODE REVIEW

**Comment by agreco** [comment]
Sweet, I've just pulled a copy of Reaction. I'll take a look at this next

File: src/ComponentRenderer.es6 [hunk 1]
```diff
@@ -15,18 +15,44 @@ class ComponentRenderer {
...
-    const Component = this.components[state.component.name];
+    const Component = this.components[state.react.component];
     const factory = React.createFactory(Component);
     return factory(state);
   }
+
+  _mergeRecursive(oldObj, newObj) {
+    for (const k in newObj) {
+      try {
+        if (typeof newObj[k] == 'object' ) {
```
Comment (agreco) [inline src/ComponentRenderer.es6:45]: Looks good, mate. 

Just a few quick questions: 
- Why the of use `==` instead of `===`
- Why the use of a try/catch

Comment (jphastings) [inline src/ComponentRenderer.es6:45]: To be perfectly honest this was a [copy/paste job](http://stackoverflow.com/a/383245/201130), but `===` seems like overkill as they're both strings; the exception catching is for the case where `newObj.key1` exists and is an object, but `oldObj` is not an object.

Comment (agreco) [inline src/ComponentRenderer.es6:45]: Cool cool, I suggest we revisit this just to tidy it up a little.. but overall looks pretty legit

**Pull request closed without merging**